clang/test/CIR/Transforms/simpl.c

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir -mmlir --mlir-print-ir-before=cir-simplify %s -o %t1.cir 2>&1 | FileCheck -check-prefix=BEFORE %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir -mmlir --mlir-print-ir-after=cir-simplify %s -o %t2.cir 2>&1 | FileCheck -check-prefix=AFTER %s
+
+
+#define CHECK_PTR(ptr)  \
+  do {                   \
```

Comment:
well, we don't have( maybe I will try to implement. From the other point of view, it kind of convenient to have `do .. while(0)` since we know where the macroses are) and may use this knowledge for some analysis.

Reply:
Good point. Right now we don't have a distinction between folders/simplifications meant for cleaning up versus optimization - MergeCleanups (now CIRSimplify) runs before all other passes (even before lifetime checker), which is still harmless because we don't do anything crazy. Looking fwd we might need to separate things that look more like canonicalization from optimization.